### DISCUSSION TITLE
Insertable

### DISCUSSION THREAD

**Discussion by jutbin** [diesel-rs/diesel#4149] [Q&A]
Which types implement the Insertable trait?

**Comment by jutbin** [comment]
Or how should I find the implementation of the Insertable trait?

**Comment by weiznich** [comment]
That's an question that's impossible to answer for the general case as any crate can implement the `Insertable` trait for it's own types. You can check locally which types in your project implement `Insertable` by performing the following steps:

* Run `cargo doc --open`
* Search for `Insertable`
* Navigate to the `diesel::Insertable` trait documentation
* Select `Implementors` in the side bar
* See a list of all types that implement this trait

**Reply by jutbin** [reply]
Can I customize the implementation of a chrono::DateTime<chrono::Local> type Insertable and Queryable? If so, how to do it? Can you give an example?

**Reply by weiznich** [reply]
It's not meaningful to customize the implementation at that level. If you really want to go down that route you should look at the [`ToSql`](https://docs.diesel.rs/2.2.x/diesel/serialize/trait.ToSql.html) and [`FromSql`](https://docs.diesel.rs/2.2.x/diesel/deserialize/trait.FromSql.html) traits, that describe how to translate a rust type to and from the database representation. Both links contain examples for custom type implementations. As you want to implement that for a type from a third party crate, you need to use a new type wrapper due to rusts coherence rules (check the generic rust documentation for new types for that).
